### DISCUSSION TITLE
The Proble function will be enabled for rotary axis A ?

### DISCUSSION THREAD

**Discussion by BUGNOTTO** [terjeio/ioSender#203] [General]
Hello, how is everything ? I'm new to the io sender program. I used to use Mach3 on my CNC.

As I work with making gold rings and rings, I miss the surface leveling on the rotating axis.

I would like to know if the Proble function will be enabled for rotary axis A in version 2.0.37?

**Comment by BUGNOTTO** [comment]
Taking advantage of the topic, I would like to suggest some way of tracking the capture of points during the execution of the problem function. Example how many points have already been scanned and when they are missing and if possible visualize the points in 3D or on the route.

**Comment by terjeio** [comment]
> I miss the surface leveling on the rotating axis.

Is this something Mach3 supports? Is it documented somewhere? Is it about Z motion vs A surface? Heightmap for AX or AY coordinates?
Or something [like this](https://github.com/tsipa/mach3-rotary-axis-probe)?
Do you have links to videos showing how it works?

> I would like to know if the Proble function will be enabled for rotary axis A in version 2.0.37?

I do not know how this function is supposed to work, so I cannot tell.

**Comment by BUGNOTTO** [comment]
Good morning my friend

So I use the G Code Ripper software with Mach3. Have you used g code ripper with mach3?

**Reply by terjeio** [reply]
> So I use the G Code Ripper software with Mach3. 

Perhaps it can be used with grblHAL as well? It is possible to compile grblHAL with variable/expression support. No subroutines though.

>  Have you used g code ripper with mach3?

No, never heard about it before. And it can be mentioned that I do not own a rotary axis fixture...

**Reply by BUGNOTTO** [reply]
> > Então eu uso o software G Code Ripper com Mach3.
> 
> Talvez possa ser usado com grblHAL também? É possível compilar grblHAL com suporte a variável/expressão. Sem sub-rotinas embora.
>  I believe it is possible to use g-code-ripper with IO Sender.
> > Você já usou o estripador de código g com mach3?
> 
> Não, nunca ouvi falar sobre isso antes. E pode-se mencionar que não possuo uma fixação de eixo rotativo...

**Comment by BUGNOTTO** [comment]
https://www.youtube.com/watch?v=p27n0LK0fm0

**Comment by BUGNOTTO** [comment]
https://www.youtube.com/watch?v=e2kMbJj3umY

**Comment by BUGNOTTO** [comment]
https://www.youtube.com/watch?v=T3H32hoKAcU

**Comment by BUGNOTTO** [comment]
these are examples of how to use mach3 with G Code Ripper. I know that an example of Auto Leveling is on the x,y axis but I do the same on the A axis

**Reply by terjeio** [reply]
> but I do the same on the A axis

So you have an _irregular_ (not a perfect cylinder) AX or AY plane you want to wrap a XAZ or YAZ design to? Or a XYZ design you want to wrap to XAZ or YAZ? Or what...?

Do you know how to write C# code and can add this yourself? Or do I have to buy a rotary and do it for you?

**Reply by terjeio** [reply]
Initial (and untested) code added to [latest edge build](https://github.com/terjeio/ioSender/discussions/176#discussioncomment-2257593).
